jaeger-core/src/main/java/com/uber/jaeger/samplers/GuaranteedThroughputSampler.java

```diff
@@ -37,11 +43,39 @@
 @EqualsAndHashCode
 @AllArgsConstructor (access = AccessLevel.PACKAGE) // Visible for testing
 public final class GuaranteedThroughputSampler implements Sampler {
-  private final ProbabilisticSampler probabilisticSampler;
-  private final RateLimitingSampler lowerBoundSampler;
+  public static final String TYPE = "lowerbound";
+
+  private ProbabilisticSampler probabilisticSampler;
```

Comment:
before, we would create a new guaranteedthroughput sampler on each peroperationsamper.update call if the strategy changed slightly (which is highly likely). By adding this update function in guaranteedthroughput sampler, we can update the 2 different constituent samplers separately. Creating a new RateLimitingSampler every update is a big nono since the samplers would sample every minute.